plugins/API/API.php

```diff
@@ -614,7 +614,9 @@ public function getSuggestedValuesForSegment($segmentName, $idSite)
             $values = $this->getSuggestedValuesForSegmentName($idSite, $segment, $maxSuggestionsToReturn);
         }
 
-        $values = $this->getMostFrequentValues($values);
+        if ($this->doesSegmentNeedMostFrequentValues($segmentName)) {
+            $values = $this->getMostFrequentValues($values);
+        }
```

Comment:
What is this new condition needed for? I guess actually that check now prevents all other segments from showing suggestions at all.

Reply:
`getMostFrequentValues()` is grouping and sorting the values list, but as the static lists have already been sorted by most frequently used and do not contain any duplicates we don't want them re-sorted alphabetically. So this check calls a new method that returns true for all segments that are not in a hard coded exclude list, which currently only contains countryName, browserName and operatingSystemName. This is similar to the way `doesSegmentNeedActionsData()` works.

I think it would be nicer if there was some way to have an `isStaticList` boolean property for the segment rather than a hard coded list of segment names, but I couldn't see a clean way to do this without having a segment class.